internal/payloads/reservation_payload.go

```diff
@@ -2,38 +2,85 @@ package payloads
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/RHEnVision/provisioning-backend/internal/models"
 	"github.com/go-chi/render"
 )
 
 // ReservationRequest is empty, account comes in HTTP header and
 // provider type in HTTP URL. All other fields are auto-generated.
-type ReservationRequest struct {
-}
 
-type ReservationResponse struct {
-	*models.Reservation
-}
+type ReservationResponsePayload struct {
+	ID int64 `json:"id"`
+
+	// Provider type. Required.
+	Provider int `json:"provider"`
+
+	// Time when reservation was made.
+	CreatedAt time.Time `json:"created_at"`
 
-type NoopReservationRequest ReservationRequest
+	// Textual status of the reservation or error when there was a failure
+	Status string `json:"status"`
 
-type NoopReservationResponse struct {
-	*models.NoopReservation
+	// Time when reservation was finished or nil when it's still processing.
+	FinishedAt *time.Time `json:"finished_at"`
+
+	// Flag indicating success, error or unknown state (NULL). See Status for the actual error.
+	Success *bool `json:"success"`
 }
 
-type AWSReservationRequest struct {
-	*models.AWSReservation
+type AWSReservationResponsePayload struct {
+	Reservation ReservationResponsePayload
```

Comment:
This is a typo - you actually add a new field named Reservation which is not marshalled because it is missing the JSON tag. Since we only want to show reservation ID, I would simply copy the ID here and copy it manually in the function, no need to make things more complex then they need. Do not embed this is what I mean.